src/AutoBuild.jl

```diff
@@ -1218,44 +1219,63 @@ function build_jll_package(src_name::String, build_version::VersionNumber, code_
-
-        ```julia
-        using $(src_name)_jll
-
-        # For file products, you can access its file location directly:
-        data_lines = open(data_txt, "r") do io
-            readlines(io)
-        end
-
-        # For library products, you can use the exported variable name in `ccall()` invocations directly
-        num_chars = ccall((libdataproc, :count_characters), Cint, (Cstring, Cint), data_lines[1], length(data_lines[1]))
-
-        # For executable products, you can use the exported variable name as a function that you can call
-        mungify_exe() do mungify_exe_path
-            run(`\$mungify_exe_path \$num_chars`)
-        end
-        ```
-        """)
+        print(io,
+              """
+              # $(src_name)_jll.jl
+
+              This is an autogenerated package constructed using [`BinaryBuilder.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl).
+
+              """,
+              iszero(length(keys(products_info))) ? "" :
+              """
+              ## Products
+
+              The code bindings within this package are autogenerated from the following `Products` defined within the `build_tarballs.jl` file that generated this package:
+
+              ```julia
+              products = [
+              $(join(collect("    $(typeof(p))($(product_names(p)), :$(variable_name(p)))" for (p, _) in products_info), ",\n"))
+              ]
+              ```
+
+              """,
+              """
+              ## Usage example
+
+              For example purposes, we will assume that the following products were defined in the imaginary package `Example_jll`:
+
+              ```julia
+              products = [
+                  FileProduct("src/data.txt", :data_txt),
+                  LibraryProduct("libdataproc", :libdataproc),
+                  ExecutableProduct("mungify", :mungify_exe)
+              ]
+              ```
+
+              With such products defined, `Example_jll` would contain `data_txt`, `libdataproc` and `mungify_exe` symbols exported. For `FileProduct` variables, the exported value is a string pointing to the location of the file on-disk.  For `LibraryProduct` variables, it is a string corresponding to the `SONAME` of the desired library (it will have already been `dlopen()`'ed, so typical `ccall()` usage applies), and for `ExecutableProduct` variables, the exported value is a function that can be called to set appropriate environment variables.  Example:
```

Comment:
Could you make the example use the library in question? i.e. base the `ccall` on the first `LibraryProduct` (or drop it if it doesn't exist), etc.

Reply:
I thought so initially, but the call would be completely made up, potentially confusing a user reading the example and trying it.

Reply:
Could you allow it to be set in the `build_tarballs.jl`? I think the benefit of having runnable examples would be worth the additional complexity.

Reply:
Some recipes build dozens of libraries; I don't think it's going to be easy to get people to give examples for all of them.  On the other hand, it would be neat to have people be able to specify them.  We could have a "example `ccall()`" parser where in the wizard we can ask you "just paste in an exemplary `ccall()` if one exists", and you can provide that ccall in the `build_tarballs.jl` as well.

Hmmm, I'm not sold, but it's an interesting idea.

Reply:
I think a single `ccall` would be fine.

Reply:
Could also just link to proper docs about how to use executable products, library products and file products.